app/src/main/java/com/portfolio/tasky/viewModels/LoginViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.portfolio.tasky.viewModels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.portfolio.tasky.usecases.EmailPatternValidatorImpl
+import com.portfolio.tasky.usecases.PasswordPatternValidatorImpl
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val validatorImpl: EmailPatternValidatorImpl,
+    private val passwordPatternValidatorImpl: PasswordPatternValidatorImpl
```

Comment:
Here you should send the abstractions as parameter, but inject the implementation in the ValidatorModule

This is useful because when testing the viewmodel, you could pass "FakeEmailValidator" or something

Reply:
Also, the names of the email validator isn't that descriptive
"validatorImpl" doesn't provide information

I'd go with something like
"emailValidator"

Alslo, try not to add "Impl" to the variable names

So for the password one, i'd fo with
"passwordValidator"

Reply:
Done